src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -477,6 +477,37 @@ private static bool TryGetClassDeclarationList(INamedTypeSymbol typeSymbol, [Not
                 return true;
             }
 
+            private static string GetClassDeclarationName(INamedTypeSymbol typeSymbol)
+            {
+                if (!typeSymbol.IsGenericType || typeSymbol.TypeArguments.Length == 0)
+                {
+                    return typeSymbol.Name;
+                }
+
+                StringBuilder sb = new StringBuilder();
+
+                sb.Append(typeSymbol.Name);
+                sb.Append('<');
+
+                bool first = true;
+                foreach (ITypeSymbol typeArg in typeSymbol.TypeArguments)
+                {
+                    if (!first)
+                    {
+                        sb.Append(", ");
+                    }
+                    else
+                    {
+                        first = false;
+                    }
+                    sb.Append(typeArg.Name);
```

Comment:
I'm curious: should we be doing this recursively? (if no, why not?) i.e. will this work for `Foo<Bar<Baz<string>>>`?

Reply:
Agreed, and nesting patterns can be fairly arbitrary in generics, e.g. `Dictionary<string, Dictionary<string, List<string>>`.  Would it be possible to add tests for that?

Reply:
Good point, I'll check it.

Reply:
This code is for a generic JsonSerializerContext descendant class declaration generation (to fix https://github.com/dotnet/runtime/issues/66880). Something like:

```
internal partial class MyJsonContext<T> : JsonSerializerContext { }
```

I can't imagine if any sort of nested generic parameters is possible here.
Anyway I'll add a test for `Dictionary<string, Dictionary<string, List<string>>` as serializable type.

Reply:
It turned out that such a configuration already exists =)

https://github.com/dotnet/runtime/blob/a2990b5e245ce08a9e9643361b79dc2f461da773/src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/Serialization/CollectionTests.cs#L96

Reply:
Also note this could be something like:

```
internal partial class MyJsonContext<Foo<T>> : JsonSerializerContext { }
```

please add a test that this works as well

Reply:
Unfortunately I can't. Such code is not compilable: https://sharplab.io/#v2:C4LglgNgPgAgTARgLAChVgHbAKYCcMCGEABAA4G7BhHHzECyAngFIDOA9hgMKc4AewADwAxdu0EAVAHxTiAb2IBfVDADMtOMVHjp8palRA==